python/lsst/afw/table/schema/schema.cc

```diff
@@ -82,6 +82,48 @@ void declareFieldBaseSpecializations(PyFieldBase<std::string> &cls) {
     cls.def("getSize", &FieldBase<std::string>::getSize);
 }
 
+// Specializations for Field
+
+template <typename T>
+void declareFieldSpecializations(PyField<T> &cls) {
+    cls.def(py::pickle(
+            [](Field<T> const &self) {
+                /* Return a tuple that fully encodes the state of the object */
+                return py::make_tuple(self.getName(), self.getDoc(), self.getUnits());
+            },
+            [](py::tuple t) {
+                if (t.size() != 3) throw std::runtime_error("Invalid number of parameters when unpickling!");
+                return Field<T>(t[0].cast<std::string>(), t[1].cast<std::string>(), t[2].cast<std::string>());
+            }));
+}
+
+template <typename U>
+void declareFieldSpecializations(PyField<Array<U>> &cls) {
```

Comment:
`U` -> `T`

Reply:
As with the `makeKey` methods, you could make a template that covers both `Array` and `string`, and without the need for an explicit template argument.